lua/bufferline/render.lua

```diff
@@ -172,6 +175,14 @@ local function render(update_names)
         buffer_data.width or
         (layout.base_widths[i] + 2 * layout.padding_width)
 
+    if has_icons and not has_numbers then
+      width = width + 2
+    elseif has_icons then
+      width = width + 1
+    end
+
+    if has_numbers then width = width + #tostring(buffer_number) + 1 end
+
```

Comment:
If `buffer_data.width` has a non-`nil` value, it means there is an ongoing animation and that `buffer_data.width` is the width that should be used for rendering.

Also this project uses `len()` instead of `#` because I'm a FP fan and I like functions very much. And also I find it more readable.